core/src/main/java/nu/yona/server/subscriptions/service/migration/EncryptBuddyLastStatusChangeTime.java

```diff
@@ -0,0 +1,19 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Stichting Yona Foundation This Source Code Form is subject to the terms of the Mozilla Public License, v.
+ * 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *******************************************************************************/
+package nu.yona.server.subscriptions.service.migration;
+
+import nu.yona.server.subscriptions.entities.User;
+import nu.yona.server.subscriptions.service.MigratePrivateUserDataService;
+
+public class EncryptBuddyLastStatusChangeTime extends MigratePrivateUserDataService.MigrationStep
+{
+	@SuppressWarnings("deprecation")
+	@Override
+	public void upgrade(User userEntity)
+	{
+		userEntity.getBuddies()
+				.forEach(buddy -> buddy.setLastStatusChangeTime(buddy.getBuddyAnonymized().getLastStatusChangeTime()));
```

Comment:
Please also nullify the time stamp on {{BuddyAnonymized}}. With that, we prevent identity matching (the purpose of this ticket) and we can see how far we are with the migration.